dashboard/app/controllers/lessons_controller.rb

```diff
@@ -17,6 +17,14 @@ def edit
       id: @lesson.id,
       name: @lesson.name,
       overview: @lesson.overview,
+      studentOverview: @lesson.student_overview,
```

Comment:
Let's pause here and think about how best to design this API. We want to avoid getting into a situation like the `ScriptsController` API, where it's hard to remember all the translations that need to happen in the parameter names between the front and backend.

Some potential goals for this API and its implementation:

- The data provided by`edit` can be directly passed back to `update`. It seems like this isn't the case right now because of the camel casing? Maybe ruby is magic enough that it handles this though.
- `update` raises an error if any unknown parameters are passed (maybe it already does this.)
- Reducing code changes / boilerplate code for new properties and columns. (At the very least, it seems like we could make `properties` require no code changes.)

This can be in a separate / followup PR, but I think it'd be valuable to do soon.

Any thoughts @davidsbailey ?

Reply:
> Some potential goals for this API and its implementation:
> 
> * The data provided by`edit` can be directly passed back to `update`. It seems like this isn't the case right now because of the camel casing? Maybe ruby is magic enough that it handles this though.

I could be mistaken but I don't think there is any ruby magic for this. If we make this a requirement, it seems like we'd need to leave the server variable names in snake case when they are handled on the client. 

> * `update` raises an error if any unknown parameters are passed (maybe it already does this.)

I think we typically use the `params.require(...).permit(...)` pattern, which I would guess ignores extraneous parameters rather than raising.

> * Reducing code changes / boilerplate code for new properties and columns. (At the very least, it seems like we could make `properties` require no code changes.)

I think this is a good goal, but it could come at the expense of the client needing to know the data structure which exists on the server, specifically which fields are properties or not.

Reply:
I wrote up my proposal here: https://docs.google.com/document/u/1/d/19_aYMPAQaCSfngZHBNH8J7NAetuz2xOsjT8FnPXTbsY/edit#

But I can address some of these points here:

> I think we typically use the params.require(...).permit(...) pattern, which I would guess ignores extraneous parameters rather than raising.

Yea, it'd be ideal if we can continue to use the existing Rails parameters support but get this behavior. It seems there's a way to do it globally: https://www.bignerdranch.com/blog/catching-strong-params-problems-early/ It'd probably break something somewhere, though.

Implementation aside, do you agree it's desirable? I've definitely wasted time debugging a value not being set when I was simply passing in the wrong parameter name.

> I think this is a good goal, but it could come at the expense of the client needing to know the data structure which exists on the server, specifically which fields are properties or not.

I agree that it should be a stronger goal to hide/decouple the server implementation details from the client. And thinking about it more, because of validation (and because shapes are not already defined in `properties`), we'd probably want to make code changes anyway. So this is a minor thing, but it's still something to keep in mind, mainly because the more manual changes needed, the easier it is to mess something up.